contracts/registry/Registry.sol

```diff
@@ -45,139 +45,96 @@ contract Registry is
-    // TODO service registration means its approval for some type?
-    // TODO registration of precompile addresses
     function register(ObjectInfo memory info)
         external
         onlyRegistryService
         returns(NftId nftId)
     {
-        ObjectType objectType = info.objectType;
-        NftId parentNftId = info.parentNftId;
-        ObjectInfo memory parentInfo = _info[parentNftId];
-        ObjectType parentType = parentInfo.objectType; // see function header
-        address parentAddress = parentInfo.objectAddress;
-
-        // parent is contract -> need to check? -> check before minting
-        // special case: global registry nft as parent when not on mainnet -> global registry address is 0
-        // special case: when parentNftId == _chainNft.mint(), check for zero parent address before mint
-        // special case: when parentNftId == _chainNft.mint() && objectAddress == initialOwner
-        if(parentAddress == address(0)) {
-            revert ZeroParentAddress();
+        // no service registrations
+        if(info.objectType == SERVICE()) {
+            revert();
         }
 
-        address interceptor = _getInterceptor(info.isInterceptor, info.objectAddress, parentInfo.isInterceptor, parentAddress);
-
-        // TODO does external call
-        uint256 mintedTokenId = _chainNft.mint(
-            info.initialOwner,
-            interceptor,
-            EMPTY_URI);
-        nftId = toNftId(mintedTokenId);
-
-        // TODO move nftId out of info struct
-        // getters by nftId -> return struct without nftId
-        // getters by address -> return nftId AND struct
-        info.nftId = nftId;
-        _info[nftId] = info;
-
-        if(info.objectAddress > address(0)) 
-        {
-            // TODO if need to add types latter -> at least call this check from registry service
-            // parent is registered + object-parent types are valid
-            if(_isValidContractCombination[objectType][parentType] == false) {
-                revert InvalidTypesCombination(objectType, parentType);
-            }
-
-            address contractAddress = info.objectAddress;
-
-            if(_nftIdByAddress[contractAddress].gtz()) { 
-                revert ContractAlreadyRegistered(contractAddress);
-            }
-
-            _nftIdByAddress[contractAddress] = nftId;
+        nftId = _register(info);
+    }
 
-            // special case
-            if(objectType == SERVICE()) {
-                _registerService(info);
-            }
-        }
-        else
-        {
-            if(_isValidObjectCombination[objectType][parentType] == false) {
-                revert InvalidTypesCombination(objectType, parentType);
-            }
+    // assume release manager keeps versioning right and activates only when release is ready to use
+    function setServiceState(VersionPart version, ObjectType serviceType, bool state)
```

Comment:
What is the meaning of `state`? Does this indicate if a service is active? If so, maybe `setServiceActive` would be a better name. Unless the state can have more states than a boolean (active, inactive).